cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/TranslationConfiguration.kt

```diff
@@ -651,171 +651,73 @@ private constructor(
-                mutableMapOf<KClass<out Pass<*>>, MutableSet<KClass<out Pass<*>>>>()
-
-            // Add the "execute before" dependencies.
-            for (p in passes) {
-                val executeBefore = mutableListOf<KClass<out Pass<*>>>()
-
-                val depAnn = p.findAnnotations<DependsOn>()
-                // collect all dependencies added by [DependsOn] annotations.
-                for (d in depAnn) {
-                    val deps =
-                        if (d.softDependency) {
-                            softDependencies.computeIfAbsent(p) { mutableSetOf() }
-                        } else {
-                            hardDependencies.computeIfAbsent(p) { mutableSetOf() }
-                        }
-                    deps += d.value
-                }
-
-                val execBeforeAnn = p.findAnnotations<ExecuteBefore>()
-                for (d in execBeforeAnn) {
-                    executeBefore.add(d.other)
-                }
-
-                for (eb in executeBefore) {
-                    passes
-                        .filter { eb == it }
-                        .forEach {
-                            val deps = softDependencies.computeIfAbsent(it) { mutableSetOf() }
-                            deps += p
-                        }
-                }
+        private fun mermaidPassName(pass: KClass<out Pass<*>>): String {
+            return when {
+                pass.isFirstPass -> "FirstPass"
+                pass.isLastPass -> "LastPass"
+                else -> pass.simpleName ?: "UnknownPass"
             }
-
-            log.info(
-                "The following mermaid graph represents the pass dependencies: \n ${buildMermaid(softDependencies, hardDependencies)}"
-            )
-
-            for (p in passes) {
-                var passFound = false
-                for ((pass) in workingList.getWorkingList()) {
-                    if (pass == p) {
-                        passFound = true
-                        break
-                    }
-                }
-                if (!passFound) {
-                    workingList.addToWorkingList(
-                        PassWithDependencies(
-                            p,
-                            softDependencies[p] ?: mutableSetOf(),
-                            hardDependencies[p] ?: mutableSetOf()
-                        )
-                    )
-                }
-            }
-            return workingList
         }
-
         /**
          * Builds a markdown representation of a pass dependency graph, based on
          * [Mermaid](https://mermaid.js.org) syntax.
          */
-        private fun buildMermaid(
-            softDependencies: MutableMap<KClass<out Pass<*>>, MutableSet<KClass<out Pass<*>>>>,
-            hardDependencies: MutableMap<KClass<out Pass<*>>, MutableSet<KClass<out Pass<*>>>>
-        ): String {
+        private fun buildMermaid(passes: List<KClass<out Pass<out Node>>>): String {
```

Comment:
This became quiet a bit of code. Should we move this to a separate file?

Reply:
Yes that would make sense

Reply:
[faef9e2](https://github.com/Fraunhofer-AISEC/cpg/pull/1645/commits/faef9e2d66970f125800e282780ae94226e22e12)